src/sentry/api/endpoints/project_transaction_threshold_override.py

```diff
@@ -0,0 +1,161 @@
+    ProjectTransactionThresholdOverride,
+)
+
+MAX_TRANSACTION_THRESHOLDS_PER_PROJECT = 100
+
+
+class ProjectTransactionThresholdOverrideSerializer(serializers.Serializer):
+    transaction = serializers.CharField(required=True, max_length=200)
+    threshold = serializers.IntegerField(required=True)
+    metric = serializers.CharField(required=True)
+
+    def validate_metric(self, metric):
+        for key, value in TRANSACTION_METRICS.items():
+            if value == metric:
+                return key
+
+        raise serializers.ValidationError(f"Invalid transaction metric - {metric}")
+
+    def validate_threshold(self, threshold):
+        if threshold % 100:
+            raise serializers.ValidationError("Invalid threshold - specify a multiple of 100")
+
+        return threshold
+
+    def validate(self, data):
+        data = super().validate(data)
+        organization = self.context.get("organization")
+        project = self.context.get("project")
+        count = (
+            ProjectTransactionThresholdOverride.objects.filter(
+                project=project, organization=organization
+            )
+            .exclude(transaction=data["transaction"])
+            .count()
+        )
+        if count >= MAX_TRANSACTION_THRESHOLDS_PER_PROJECT:
+            raise serializers.ValidationError(
+                f"At most {MAX_TRANSACTION_THRESHOLDS_PER_PROJECT} configured transaction thresholds per project."
+            )
+
+        return data
+
+
+class ProjectTransactionThresholdOverrideEndpoint(OrganizationEventsV2EndpointBase):
+    permission_classes = (ProjectTransactionThresholdOverridePermission,)
+
+    def has_feature(self, organization, request):
+        return features.has(
+            "organizations:project-transaction-threshold-override",
+            organization,
+            actor=request.user,
+        )
+
+    def get_project(self, request, organization):
+        projects = self.get_projects(request, organization)
+        if len(projects) != 1:
+            raise ParseError("Only 1 project per transaction threshold")
+
+        return projects[0]
+
+    def get(self, request, organization):
+        if not self.has_feature(organization, request):
+            return self.respond(status=status.HTTP_404_NOT_FOUND)
+
+        project = self.get_project(request, organization)
+
+        try:
+            project_threshold = ProjectTransactionThresholdOverride.objects.get(
+                transaction=request.GET.get("transaction"),
+                project_id=project.id,
+                organization_id=organization.id,
+            )
+        except ProjectTransactionThresholdOverride.DoesNotExist:
+            return Response(
+                data={
+                    "transaction": request.GET.get("transaction"),
+                    "projectId": str(project.id),
+                    **DEFAULT_THRESHOLD,
+                },
+                status=status.HTTP_200_OK,
+            )
```

Comment:
I wonder if it makes sense to create a `ProjectTransactionThresholdOverride` without writing it to DB and let the serializer handle it so that the response is consistent.